src/rt-app_taskgroups.c

```diff
@@ -253,6 +268,60 @@ static int cgroup_mkdir(const char *name, int *offset)
 		} else if (*offset == -1) {
 			*offset = y;
 		}
+
+		if (ctrl.version == 2) {
+			size_t path_size = strlen(path) + strlen(SUBTREE_CONTROL) + 2;
+			char* cgroup_path = malloc(path_size);
+			FILE *cgroup_file;
+			snprintf(cgroup_path, path_size, "%s/%s", path, SUBTREE_CONTROL);	
+			cgroup_file = fopen(cgroup_path, "we");
+			if (!cgroup_file) {
+				perror("fopen");
+				goto error;
+			}
+			fprintf(cgroup_file, "+cpu");
+			if (fclose(cgroup_file)) {
+				perror("fclose");
+				goto error;
+			}
+			free(cgroup_path);
+			
+			path_size = strlen(path) + strlen(TYPE) + 2;
+			cgroup_path = malloc(path_size);
+			snprintf(cgroup_path, path_size, "%s/%s", path, TYPE);
+			cgroup_file = fopen(cgroup_path, "we");
+			if (!cgroup_file) {
+				perror("fopen");
+				goto error;
+			}	
+			if (fprintf(cgroup_file, "threaded") < 0) {
+				perror("fprintf");
+				goto error;
+			}
+			if (fclose(cgroup_file)) {
+				perror("fclose");
+				goto error;
+			}
+			free(cgroup_path);
+
+			path_size = strlen(path) + strlen(PROCS) + 2;
+			cgroup_path = malloc(path_size);
+			snprintf(cgroup_path, path_size, "%s/%s", path, PROCS);
+			cgroup_file = fopen(cgroup_path, "we");
+			if (!cgroup_file) {
+				perror("fopen");
+				goto error;
+			}
+			if (fprintf(cgroup_file, "%d", getpid()) < 0) {
```

Comment:
Why are you putting the full rt-app process in the newly created group ? The end result is that all threads will be in the last created cgroup by default even those which don't set a "taskgroup" field

Reply:
Additionally, wouldn't this drag the process through all created groups (top-down)?